lib/models/node.js

```diff
@@ -82,28 +83,34 @@ function NodeModelFactory (Model, waterline, _, Promise, Constants) {
             }
         },
         addTags: function(id, tags) {
-            // TODO: Refactor to native mongo command once it is merged
-            return Promise.fromNode(this.native.bind(this))
-                .then(function(collection) {
-                    return collection.update(
-                        { _id: waterline.nodes.mongo.objectId(id) },
-                        {
+            var self = this;
+            var query = { _id: waterline.nodes.mongo.objectId(id) };
+            var update = {
                             $addToSet: { tags: { $each: tags } },
                             $set: { updatedAt: new Date() }
-                        });
-                });
+                        };
+            var options = { new: true };
+            return self.findAndModifyMongo(query, {}, update, options)
+            .then(function(record) {
+                record.id = record._id;
+                waterlineProtocol.publishRecord(self, 'updated', record)
+                return record;
+            });
         },
         remTags: function(id, tag) {
-            // TODO: Refactor to native mongo command once it is merged
-            return Promise.fromNode(this.native.bind(this))
-                .then(function(collection) {
-                    return collection.update(
-                        { _id: waterline.nodes.mongo.objectId(id) },
-                        {
+            var self = this;
+            var query = { _id: waterline.nodes.mongo.objectId(id) };
+            var update = {
                             $pull: { tags: tag },
                             $set: { updatedAt: new Date() }
-                        });
-                });
+                        };
+            var options = { new: true };
+            return self.findAndModifyMongo(query, {}, update, options)
+            .then(function(record) {
+                record.id = record._id;
+                waterlineProtocol.publishRecord(self, 'updated', record)
```

Comment:
Wonder if we should just move the publishRecord code into the `findAndModifyMongo` function?